api/accessrequest/access_request_test.go

```diff
@@ -125,3 +126,39 @@ func TestGetResourceDetails(t *testing.T) {
 	// This Okta sourced user group should have a friendly name.
 	require.Equal(t, "friendly group 1", details[types.ResourceIDToString(resourceIDs[4])].FriendlyName)
 }
+
+func TestGetResourceNames(t *testing.T) {
+	clusterName := "cluster"
+
+	presence := &mockResourceLister{
+		resources: []types.ResourceWithLabels{
+			newNode(t, "node1", "hostname 1"),
+			newApp(t, "app1", "friendly app 1", types.OriginDynamic),
+			newApp(t, "app2", "friendly app 2", types.OriginDynamic),
+			newApp(t, "app3", "friendly app 3", types.OriginOkta),
+			newUserGroup(t, "group1", "friendly group 1", types.OriginOkta),
+		},
+	}
+	resourceIDs := []types.ResourceID{
+		newResourceID(clusterName, types.KindNode, "node1"),
+		newResourceID(clusterName, types.KindApp, "app1"),
+		newResourceID(clusterName, types.KindApp, "app2"),
+		newResourceID(clusterName, types.KindApp, "app3"),
+		newResourceID(clusterName, types.KindUserGroup, "group1"),
+	}
+
+	ctx := context.Background()
+
+	req, err := types.NewAccessRequestWithResources(uuid.New().String(), "some-user", []string{}, resourceIDs)
+	require.NoError(t, err)
+	names, err := GetResourceNames(ctx, presence, req)
+	require.NoError(t, err)
+	expected := []string{
+		"node/hostname 1",
+		"/cluster/app/app1",
+		"/cluster/app/app2",
+		"app/friendly app 3",
+		"user_group/friendly group 1",
+	}
```

Comment:
can you tell me the reasoning behind some resources having the name prepended and some not?
for example in my request
```
  "resource_names": [
    "/avatus.sh/app/avatus.spaces",
    "/avatus.sh/app/client",
    "/avatus.sh/kube_cluster/testkubecluster",
    "node/avatus",
    "/avatus.sh/windows_desktop/windows-server-2019"
  ],
```

Reply:
They are only prepended if the source details has a 'friendly' name set.

Reply:
I had the same question. Looks like [ResourceIDToString](https://github.com/gravitational/teleport/blob/ccbf2dfcddbeb7f91c2257d88027664ea54ac8da/api/types/resource_ids.go#L72-L78) prepends the leading `/`. Should we do the same when constructing the name from the friendly name for consistency?